src/compiler/crystal/codegen/link.cr

```diff
@@ -158,7 +158,11 @@ module Crystal
       # Add CRYSTAL_LIBRARY_PATH locations, so the linker preferentially
       # searches user-given library paths.
       CrystalLibraryPath.paths.each do |path|
-        flags << Process.quote_posix("-L#{path}")
+        # we use `quote` rather than `quote_posix` because these flags are
+        # evaluated on the compiler host system and Windows rules could be used
+        # for MinGW-w64
+        # TODO: what about the opposite? (e.g. MSVC on Wine on Linux)
+        flags << Process.quote("-L#{path}")
```

Comment:
**thought:** Expecting the linker to run on the host system should usually be fine. Not sure about the MSVC on Wine thing. I guess we cannot really now all the details about the system environment.

But an exception is probably explicit cross-compiling. With `--cross-compile` I believe the command should be for formatted for the target platform, not the host. We cannot know for certain on which platform the linker command will be executed, but most likely it'll be on the target platform.

Reply:
I believe this applies to the rest of `Crystal::Compiler#linker_command` too, so I think we can follow that up in a similar manner